cTraining/cTraining/CalcCoins.c

```diff
@@ -0,0 +1,90 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+//CalcCoins.c
+//Implemented a method to store the number of coins in each denomination (Rs. 10, Rs. 5, Rs. 2, Rs. 1) for change.
+// ------------------------------------------------------------------------------------------------
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+/// <summary>Returns true if the input amount is Valid; otherwise, false.</summary>
+bool IsValidAmount (char* amount);
+
+/// <summary>Returns 1 if coins are counted and stored in the array; otherwise, 0.</summary>
+int CalcCoins (int paid, int actualAmount, int coinsCount[], int coins[]);
+
+/// <summary>Get the paid and actual amounts from the user and display the number of coins 
+///needed for each denomination for change.</summary>
+void PrintBalanceCoins ();
+
+/// <summary>Gets the amount from the user.</summary>
+void GetAmount (int* paid, int* actualAmount);
+
+/// <summary>To automatically test the CalcCoins method.</summary>
+int AutomationTest ();
+
+int main () {
+   PrintBalanceCoins ();
+   //AutomationTest ();
+}
+
+void GetAmount (int* paid, int* actualAmount) {
+   char paidStr[11], actualAmountStr[11]; //Assuming paid and bill amount has 11 characters.
+   do {
+      printf ("You enter the payment 1 to 2,147,483,647 \nPaid amount : ");
+      fgets (paidStr, 11, stdin);
+      printf ("Actual amount:");
+      fgets (actualAmountStr, 11, stdin);
+   } while (!(IsValidAmount (paidStr) && IsValidAmount (actualAmountStr)));

```

Comment:
This loop should continue if either of the amounts is not valid. The condition will cause errors.